src/Ocean/HydrostaticBoussinesq/HydrostaticBoussinesqModel.jl

```diff
@@ -633,9 +635,9 @@ function update_auxiliary_state_gradient!(
     reverse_indefinite_stack_integral!(dg, m, Q, A, t, elems) # top -> bottom
 
     # project w(z=0) down the stack
-    # Need to be consistent with vars_state_auxiliary
-    # A[1] = w, A[3] = wz0
-    copy_stack_field_down!(dg, m, A, 1, 3, elems)
+    index_w = varsindex(vars_state_auxiliary(m, FT), :w)
+    index_wz0 = varsindex(vars_state_auxiliary(m, FT), :wz0)
```

Comment:
The easy fix
```suggestion
    index_w = varsindex(vars_state_auxiliary(m, FT), :w)[1]
    index_wz0 = varsindex(vars_state_auxiliary(m, FT), :wz0)[1]
```

Reply:
As I discussed with @blallen, we could also change the `kernel_copy_stack_field_down` to work with ranges